instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/_server.py

```diff
@@ -136,74 +192,66 @@ def get_from_grpc_metadata(metadata, key) -> List[str]:
         else:
             yield
 
-    def _start_span(self, method):
-        span = self._tracer.start_as_current_span(
-            name=method, kind=trace.SpanKind.SERVER
-        )
-        return span
-
-    def intercept_unary(self, request, servicer_context, server_info, handler):
-
-        with self._set_remote_context(servicer_context):
-            with self._start_span(server_info.full_method) as span:
-                rpc_info = RpcInfo(
-                    full_method=server_info.full_method,
-                    metadata=servicer_context.invocation_metadata(),
-                    timeout=servicer_context.time_remaining(),
-                    request=request,
-                )
-                servicer_context = _OpenTelemetryServicerContext(
-                    servicer_context, span
-                )
-                response = handler(request, servicer_context)
-
-                _check_error_code(span, servicer_context, rpc_info)
-
-                rpc_info.response = response
-
-                return response
-
-    # For RPCs that stream responses, the result can be a generator. To record
-    # the span across the generated responses and detect any errors, we wrap
-    # the result in a new generator that yields the response values.
-    def _intercept_server_stream(
-        self, request_or_iterator, servicer_context, server_info, handler
-    ):
-        with self._set_remote_context(servicer_context):
-            with self._start_span(server_info.full_method) as span:
-                rpc_info = RpcInfo(
-                    full_method=server_info.full_method,
-                    metadata=servicer_context.invocation_metadata(),
-                    timeout=servicer_context.time_remaining(),
-                )
-                if not server_info.is_client_stream:
-                    rpc_info.request = request_or_iterator
-                servicer_context = _OpenTelemetryServicerContext(
-                    servicer_context, span
-                )
-                result = handler(request_or_iterator, servicer_context)
-                for response in result:
-                    yield response
-                _check_error_code(span, servicer_context, rpc_info)
-
-    def intercept_stream(
-        self, request_or_iterator, servicer_context, server_info, handler
-    ):
-        if server_info.is_server_stream:
-            return self._intercept_server_stream(
-                request_or_iterator, servicer_context, server_info, handler
+    def _start_span(self, handler_call_details, context):
+
+        attributes = {
+            "grpc.method": handler_call_details.method,
+        }
+
+        metadata = dict(context.invocation_metadata())
+        if "user-agent" in metadata:
+            attributes["grpc.user_agent"] = metadata["user-agent"]
```

Comment:
Hmm I'm not sure about this one; going off of [this](https://github.com/grpc/grpc/blob/1370ea77bd442896e5dffad3fb8436579c17f4fe/doc/PROTOCOL-HTTP2.md#user-agents), it looks like this is the underlying HTTP user agent header but I don't think it's appropriate to record as [`http.user_agent`](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#common-attributes).

I think `rpc.user_agent` would be ok for now, any thoughts?

Reply:
I was torn on this one, and was hoping for some advice, since I didn't really find any convention, but based on your other suggestions, perhaps `rpc` is indeed the right thing, and if we've set `rpc.system` as you suggested, we'll know it's gRPC involved here.